### PR TITLE
Firmward added, readme updated.

### PR DESCRIPTION
**Pull request by Herleybob**
Added firmware for GD32F130K6. Also updated the readme a little to note any special connections and a description about them.

### CODE REVIEW

**Comment by Herleybob** [comment]
This is still a work in progress and a port from the original Niklas Fauth firmware. It does not read current (and does not shut off based on current, etc), I believe board temp and voltage is working correctly. I am running this on a 40v battery with no issues. (41.11v fully charged). I need to add a picture of the board and the pinouts, but as it sits when you compile it, it will run the motor up to 120RPM and hold that RPM. ADC conversions are working. The current firmware supports both the main(no buzzer) and aux board(buzzer). Set definition in the config file.

What is needed is either communication between the two boards, or seperate adc inputs per board. But as of right now basic motor control is working.

**Comment by JRomainG** [comment]
This is pretty amazing, thanks for the contribution!
As it's still a work in progress I won't merge it right away, but I'm thrilled to test it out.

**Comment by JRomainG** [comment]
@Herleybob I downloaded and was able to flash the code from your fork on both boards after some trial and error. However, I wasn't able to get the motors running.

Did you have to do any specific wiring to get the "basic" behavior where the motors simply hold 120 RPM? Perhaps you have your diagram handy?

**Comment by Herleybob** [comment]
> @Herleybob I downloaded and was able to flash the code from your fork on both boards after some trial and error. However, I wasn't able to get the motors running.
> 
> 
> 
> Did you have to do any specific wiring to get the "basic" behavior where the motors simply hold 120 RPM? Perhaps you have your diagram handy?

Please explain the flashing errors you had to remedy. As far as motor operation, does the auxiliary board beep upon power up? This indicates the dma is working properly/flash was successful. Did you swap the 4-pin connector as stated in the bottom of the readme? This shouldn't affect operation though.

I do not have the diagrams yet, as i haven't worked on this since the last update. Do the motors attempt to move at all(if the board beeps upon power up).

**Comment by JRomainG** [comment]
I had a couple of issues when flashing:

* I originally tried with a different computer from usual which runs Windows, installing all the right dependecies in the right versions to get openocd / stlink / make to work was kind of a pain. I just switched back to UNIX to make everything easier.
* My version of st-link seemed to have a bug causing the flashing to fail, I checked out some issues from the [original repo](https://github.com/stlink-org/stlink) and downloaded the version from the dev branch and recompiled it, which seems to have fixed it.
* When flashing, I have a rather erratic behavior: it sometimes fails without any meaningful error, I had to try `make unlock` and `make flash`multiple times before getting it to work.

After flashing the auxiliary board, I do get the buzzer beep at start, it seems to behave like I expect it to based on my understanding of the code. However, once plugged back into the board, the beeping sound no longer occurs at startup.

Basically, after flashing the code it seems fine, but after unplugging it and plugging it back it in no longer works for some reason. I thought I may have flashed the wrong firmware on the wrong board, but I did make sure multiple times, and the buzzer does work the first time.

I switched the wiring like mentioned in the readme, and also tried with the original wiring just to check if I could at least get the beep or the motors to work, but no luck on that part. No too sure what I did wrong at this point, or what could explain the behavior that I'm seeing...
